conda_rpms/install.py

```diff
@@ -399,6 +404,195 @@ def symlink_conda(prefix, root_dir):
+
+    """
+    pyver_string = python_major_minor_version.replace('.', '')
+    if pyver_string.startswith('2'):
+        return py_path + 'c'
+    else:
+        directory, py_file = split(py_path)
+        basename_root, extension = splitext(py_file)
+        pyc_file = "__pycache__/%s.cpython-%s%sc" % (basename_root, pyver_string, extension)
+        return "%s/%s" % (directory, pyc_file) if directory else pyc_file
+
+
+def get_python_noarch_target_path(source_short_path, target_site_packages_short_path):
+    """
+    Modified version of conda/common/path.py:get_python_noarch_target_path at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    Modifications included:
+        * removed windows specific support
+
+    """
+    if source_short_path.startswith('site-packages/'):
+        sp_dir = target_site_packages_short_path
+        return source_short_path.replace('site-packages', sp_dir, 1)
+    elif source_short_path.startswith('python-scripts/'):
+        bin_dir = 'bin'
+        return source_short_path.replace('python-scripts', bin_dir, 1)
+    else:
+        return source_short_path
+
+
+def compile_pyc(python_exe_full_path, py_full_path, pyc_full_path):
+    """
+    Modified version of conda/gateways/disk/create.py:compile_pyc at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    Modification included:
+        * removed check that the pyc file already exists prior to compilation
+        * changed log.trace -> log.info (log.trace not supported)
+
+    """
+    command = [python_exe_full_path, '-Wi', '-m', 'py_compile', py_full_path]
+    subprocess.call(command)
+
+    if not isfile(pyc_full_path):
+        log.info('{} failed to compile correctly'.format(pyc_full_path))
+        return None
+
+    return pyc_full_path
+
+
+def parse_entry_point_def(ep_definition):
+    """
+    Copy of conda/core.path.py:parse_entry_point_def at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    """
+    cmd_mod, func = ep_definition.rsplit(':', 1)
+    command, module = cmd_mod.rsplit("=", 1)
+    command, module, func = command.strip(), module.strip(), func.strip()
+    return command, module, func
+
+
+def make_executable(path):
+    """
+    Modified version of conda/gateways/disk/permissions.py:make_executable at
+    be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    Modifications included:
+        * changed lstat -> os.lstat
+        * changed chmod -> os.chmod
+        * changed log.trace -> log.info (log.trace not supported)
+
+    """
+    if isfile(path):
+        mode = os.lstat(path).st_mode
+        log.info('chmod +x %s', path)
+        os.chmod(path, S_IMODE(mode) | S_IXUSR | S_IXGRP | S_IXOTH)
+    else:
+        log.error("Cannot make path '%s' executable", path)
+
+
+def replace_long_shebang(data):
+    """
+    Modified version of conda/core/portability.py:replace_long_shebang
+    at be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe.
+
+    Modifications included:
+        * removed mode check for non-binary shebang
```

Comment:
It seemed a bit pointless. `mode` is included in the original function because the function would have been reused in lots of places. 
In this `install.py` script, this function is only called by `create_python_entry_point` and [if you look at the original function](https://github.com/conda/conda/blob/be8c08c083f4d5e05b06bd2689d2cd0d410c2ffe/conda/gateways/disk/create.py#L87), in the call to `create_python_entry_point` it sets the mode to be `FileMode.text` so there just doesn't seem to be much point